plugins/single_plugins/ipc-rules.cpp

```diff
@@ -509,7 +577,9 @@ class ipc_rules_t : public wf::plugin_interface_t, public wf::per_output_tracker
         description["fullscreen"]  = toplevel ? toplevel->pending_fullscreen() : false;
         description["minimized"]   = toplevel ? toplevel->minimized : false;
         description["activated"]   = toplevel ? toplevel->activated : false;
-        description["focusable"]   = view->is_focusable();
+        description["sticky"]    = toplevel ? toplevel->sticky : false;
+        description["wset-id"]   = toplevel && toplevel->get_wset() ? toplevel->get_wset()->get_id() : -1;
```

Comment:
wsets also have their internal indices that we can access by get_wset()->get_index(). The id()s are global IDs for all 'objects' - so views, outputs, wsets all have IDs in the same ID space. But wset->get_index() is special for wsets, and they are usually the indices displayed in the overlay of wsets.

In addition, the get_id() can change for the 'same' workspace set as seen by wsets: for example, if we create wset 3, it becomes empty and we delete it, then at some point go there again, we will get a new object with the same index.

So I think that maybe for IPC clients the index should be used instead of id (not just here, but in all calls), what do you think?

Reply:
Sounds good to me!